src/geos_functions.jl

```diff
@@ -324,6 +324,9 @@ end
 buffer(ptr::GEOSGeom, width::Real, quadsegs::Integer=8, context::GEOSContext = _context) =
     GEOSBuffer_r(context.ptr, ptr, width, Int32(quadsegs))
 
+bufferWithStyle(ptr::GEOSGeom, width::Real, quadsegs::Integer=8, endCapStyle::Integer=1, joinStyle::Integer=1, mitreLimit::Real=5.0, context::GEOSContext = _context) =
+    GEOSBufferWithStyle_r(context.ptr, ptr, width, Int32(quadsegs), Int32(endCapStyle), Int32(joinStyle), mitreLimit)
+
 # enum GEOSBufCapStyles
 # enum GEOSBufJoinStyles
```

Comment:
Could you perhaps add these as well, and use them instead of integers as input argument type for bufferWithStyle?
That way it will be easier for users to find out about the different possible styles.

See for instance this example:
https://github.com/JuliaGeo/GDAL.jl/blob/38ccd2d37664673b81b1a86e9947bd5fe9a2f6d2/src/common.jl#L102
https://github.com/JuliaGeo/GDAL.jl/blob/017bf6b8492dcd2186ced297076b283c3591d798/src/gdal_h.jl#L14